### ISSUE TITLE
Add support for conditions

### ISSUE THREAD

**Issue by drewbutlerbb4** [kubeflow/kfp-tekton#33]
Add support for condition resources: https://github.com/tektoncd/pipeline/blob/master/docs/conditions.md

A direct example in kfp: https://github.com/kubeflow/pipelines/blob/0.2.2/samples/core/condition/condition.py

**Comment by drewbutlerbb4** [comment]
This issue will remain after #84 because conditions that receive parameters from task outputs still fail. Once [this issue](https://github.com/tektoncd/pipeline/issues/2336#issuecomment-611007302) is fixed another PR will be made to reconcile.
